staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -2259,6 +2260,130 @@ func TestCELMaxRecursionDepth(t *testing.T) {
 	}
 }
 
+func TestMessageExpression(t *testing.T) {
+	klog.LogToStderr(false)
+	defer klog.LogToStderr(true)
+	tests := []struct {
+		name                    string
+		costBudget              int64
+		message                 string
+		messageExpression       string
+		expectedLogErr          string
+		expectedValidationErr   string
+		expectedRemainingBudget int64
+	}{
+		{
+			name:                    "no cost error expected",
+			messageExpression:       `"static string"`,
+			expectedValidationErr:   "static string",
+			costBudget:              300,
+			expectedRemainingBudget: 300,
+		},
+		{
+			name:                  "messageExpression takes precedence over message",
+			message:               "invisible",
+			messageExpression:     `"this is messageExpression"`,
+			costBudget:            RuntimeCELCostBudget,
+			expectedValidationErr: "this is messageExpression",
+		},
+		{
+			name:                    "message used instead of messageExpression in case of syntax errors",
+			message:                 "message not messageExpression",
+			messageExpression:       `"unclosed string`,
```

Comment:
What case is being tested here? If parse fails, the messageExpression cannot be written to the CRD, right?

Reply:
You're right, we won't get this far - I've removed the test.